---
title: "Proteogenomics exploration"
output:
  rmarkdown::github_document:
    html_preview: false
---

```{r opts, echo=FALSE}
knitr::opts_chunk$set(
  fig.path = "md_files/exploration/"
)

```

```{r setup}
library(tidyverse)
library(Biobase)
library(ggthemes)
library(data.table)

theme_set(theme_tufte())

setwd("~/Documents/githubspot/deepmeltome/R")

sourceDir <- function(path, ...) {
  for (nm in list.files(path, pattern = "\\.[RrSsQq]$")) {
    source(file.path(path, nm), ...)
  }
}
sourceDir(path = file.path("pepnet"))


```

```{r IL_manual_start_PSMs} 
setwd("~/Documents/githubspot/deepmeltome/R")

#load in from kantele novel peptide search
psm_all <- read.delim("/Users/lab/Desktop/RJ_ALL_meltome_NovelpeptideSearch_ILcopy_Nov232020/target_psmtable.txt", sep = "\t")
#peptides <- readRDS(file = "~/Downloads/peptides (2).RDS") #From DeepMeltome_ALL_Nils>deepmeltome>proteoform_detection_phospho_novel_all_strips

#just pep sequence
psm_all$Peptide_AA <- gsub("[0-9+.]", "", psm_all$Peptide)

#load in cell line Husen analysis (Oct 2020, from proteogenomiccs folder on teams)
psm_husen <- read.delim("/Users/lab/Documents/githubspot/deepmeltome/data/Meltome_Set1-11_Novelpeptides_ENS99/novel_filteredBlastSpecAI_normalized_final_psmtable (1).tsv", sep = "\t")

psm_husen$Peptide_AA <- gsub("[0-9+.]", "", psm_husen$Peptide)

#in_eset <- psm_husen[gsub("[0-9+.]", "", rownames(peptides@assayData$exprs)) %in% psm_husen$Peptide_AA,]
novel_psm_all <- psm_all[psm_all$Peptide_AA %in% psm_husen$Peptide_AA,]
saveRDS(novel_psm_all, file = "~/Documents/githubspot/deepmeltome/novel_psm_all.RDS")

husen_meta_meltomedetected <- psm_husen[psm_husen$Peptide_AA %in% unique(novel_psm_all$Peptide_AA),]

rm(novel_psm_all)
gc()

#just look at ones mapped to a putative protein similarity
husen_meta_meltomedetected <- husen_meta_meltomedetected[husen_meta_meltomedetected$blastp_match != "NA",]

psms_eset <-import_psms_from_nf("/Users/lab/Desktop/RJ_ALL_meltome_NovelpeptideSearch_ILcopy_Nov232020/target_psmtable.txt", id_col = "Gene Name", peptide_col = "Peptide", protein_id_col = "Protein", quan_regex = "tmt16plex")

peptides_novel <- psms_to_peptides(psms = psms_eset, summarise_fun = sum,
                                 sample_meta_file = file.path("meta", "meltome_sample_meta.txt"),
                                 sample_id_col = "sample_id")

peptides_novel_vsn <- vsn_normalize_by_temperature(e_set = peptides_novel)
peptides_novel_normalized <- build_ratios_to_lowest_temperature(e_set = peptides_novel_vsn, sample_col = "sample_name")
peptides_novel_normalized@featureData@data$Peptide_AA <- gsub("[0-9+.]", "", peptides_novel_normalized@featureData@data$peptide)
peptides_novel_normalized@featureData@data$Novel <- peptides_novel_normalized@featureData@data$Peptide_AA %in% husen_meta_meltomedetected$Peptide_AA

fData(peptides_novel_normalized) <- cbind(fData(peptides_novel_normalized),"old_rownames" = rownames(peptides_novel_normalized))
rownames(peptides_novel_normalized) <- paste0("peptide", 1:nrow(peptides_novel_normalized))


```
```{r SaveAndReadIn!!!}
saveRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/husen_meta_meltomedetected.RDS",husen_meta_meltomedetected)
saveRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/peptides_novel.RDS",peptides_novel)
saveRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/peptides_novel_normalized.RDS",peptides_novel_normalized)
saveRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/peptides_novel_vsn.RDS",peptides_novel_vsn)
saveRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/psms_eset.RDS",psms_eset)
saveRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/psm_husen.RDS",psm_husen)
```


```{r SaveAndReadIn!!!}
husen_meta_meltomedetected <- readRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/husen_meta_meltomedetected.RDS")
peptides_novel <- readRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/peptides_novel.RDS")
peptides_novel_normalized <- readRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/peptides_novel_normalized.RDS")
peptides_novel_vsn <- readRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/peptides_novel_vsn.RDS")
psms_eset <- readRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/psms_eset.RDS")
psm_husen <- readRDS(file = "~/Documents/githubspot/clinical_ALL_2020/deepmeltome/data/psm_husen.RDS")
```

```{r DOK3_Novel}
#DOK3 aka XP_016865372.1 novel peptide: LLELDQVEGTGRPDPQILLGTPGVSLLICK
plot_id_symbol_melting(id_symbol = "DOK3", e_set = peptides_novel_normalized, additional_peptides = c("LLELDQVEGTGRPDPQILLGTPGVSLLICK"), add_splines = FALSE, label_proteins = TRUE, highlight_additional = TRUE)

```
```{r DOK3_Novel_RCHACV}
#DOK3 aka XP_016865372.1 novel peptide: LLELDQVEGTGRPDPQILLGTPGVSLLICK 
#RCHACV has a Frame shift deletion validated in depmap: Damaging, T1859fs
plot_id_symbol_melting(id_symbol = "DOK3", e_set = peptides_novel_normalized, additional_peptides = c("LLELDQVEGTGRPDPQILLGTPGVSLLICK"), add_splines = FALSE, label_proteins = TRUE, highlight_additional = TRUE, subset_cell_line = "RCH-ACV")

```


```{r DOK3_Just_Novel}
#DOK3 aka XP_016865372.1 novel peptide: LLELDQVEGTGRPDPQILLGTPGVSLLICK
plot_id_symbol_melting(id_symbol = "Nothing matches this character string", e_set = peptides_novel_normalized, additional_peptides = c("LLELDQVEGTGRPDPQILLGTPGVSLLICK"), add_splines = TRUE, label_proteins = TRUE, highlight_additional = FALSE)

```


```{r DOK3_By_Protein}
#DOK3 aka XP_016865372.1 novel peptide: LLELDQVEGTGRPDPQILLGTPGVSLLICK
plot_id_symbol_melting(id_symbol = "DOK3", e_set = peptides_novel_normalized, additional_peptides = c("LLELDQVEGTGRPDPQILLGTPGVSLLICK"), add_splines = TRUE, label_proteins = TRUE)

```
Look for validated mutations that are detected in the neoantigen proteomics

```{r KOPN8_Mutations}
#write_csv(as.data.frame(husen_meta_meltomedetected$blastp_match), file = "~/Downloads/husen_blastmatches.csv")
#convert these using gProfiler to gene symbols
gene_symbol_key <- read.delim("~/Downloads/gProfiler_hsapiens_17-12-2021_17-10-45.csv", sep = ",")
#filter to unique identifier pairs only
gene_symbol_key$identifier <- paste0(gene_symbol_key$initial_alias, ";;;", gene_symbol_key$converted_alias)
gene_symbol_key <- unique(gene_symbol_key$identifier) %>% str_split(pattern = ";;;") %>% as.data.frame()
gene_symbol_key <- as.data.frame(t(gene_symbol_key))
colnames(gene_symbol_key) <- c("blastp_match", "Gene")
#read in known mutations from depmap
KOPN8_muts_depmap <- read.delim("~/Downloads/KOPN8 mutations.csv", sep = ",")

gene_symbol_mutations <- left_join(x = KOPN8_muts_depmap, y = gene_symbol_key, by = "Gene")
husen_meta_kopn8_mutated <- husen_meta_meltomedetected[husen_meta_meltomedetected$blastp_match %in% gene_symbol_mutations$blastp_match,] 
husen_meta_kopn8_mutated <- husen_meta_kopn8_mutated[!is.na(husen_meta_kopn8_mutated$blastp_match),] %>%
  left_join(y = gene_symbol_mutations, by = "blastp_match")

```

Not exactly the same, but quite close to the damaging nonsense deletion, S* insertion
```{r KOPN8_Mutations_Plots_PRKDC}
library(ggpubr)
plot1 <- plot_id_symbol_melting(id_symbol = "PRKDC", e_set = peptides_novel_normalized, additional_peptides = c("EQMYNILMWIHDNYSRDPESETDNDSQEIFK"), add_splines = FALSE, label_proteins = TRUE, highlight_additional = TRUE, subset_cell_line = "KOPN-8")
plot1 + theme_pubr()

#too many peptides to view
# plot_id_symbol_melting(id_symbol = "PRKDC", e_set = peptides_novel_normalized, additional_peptides = c("EQMYNILMWIHDNYSRDPESETDNDSQEIFK"), add_splines = TRUE, label_proteins = TRUE, highlight_additional = TRUE)
```

